### PR TITLE
treewide: add thingy91x to sample.yaml

### PR DESCRIPTION
**Pull request by maxd-nordic**
This commit adds the thingy91x to samples that should be supported by launch.

### CODE REVIEW

File: samples/cellular/modem_shell/sample.yaml [hunk 1]
```diff
@@ -8,10 +8,12 @@ tests:
       - nrf9151dk/nrf9151/ns
       - nrf9160dk/nrf9160/ns
       - nrf9161dk/nrf9161/ns
+      - thingy91x/nrf9151/ns
     platform_allow:
       - nrf9151dk/nrf9151/ns
       - nrf9160dk/nrf9160/ns
       - nrf9161dk/nrf9161/ns
+      - thingy91x/nrf9151/ns
     tags: ci_build sysbuild
```
Comment (trantanen) [inline samples/cellular/modem_shell/sample.yaml:17]: This CI failure is seen in other modem shell PRs too (no idea yet why SNIPPET="nrf91-modem-trace-uart" doesn't work with thingy91x and why it has got into main in the first place):
`/jenkins_workspace/workspace/latest_sdk-nrf_PR-15685/twister-build/thingy91x_nrf9151_ns/nrf/samples/cellular/modem_shell/sample.cellular.modem_shell.thingy91_modem_uart_trace/b0/../../../../../../../../zephyr/kernel/timeout.c:80: undefined reference to 'sys_clock_elapsed'`

**Review by MarkusLassila — APPROVED** [review]
SLM OK

File: samples/cellular/modem_shell/sample.yaml (hunk 1 above)
Comment (trantanen) [reply samples/cellular/modem_shell/sample.yaml:17]: I've resolved this and will submit the fix to use SNIPPETS correctly with sysbuild.
PR where this is included is #15225.

File: samples/cellular/modem_shell/sample.yaml [hunk 2]
```diff
@@ -8,10 +8,12 @@ tests:
       - nrf9151dk/nrf9151/ns
       - nrf9160dk/nrf9160/ns
       - nrf9161dk/nrf9161/ns
+      - thingy91x/nrf9151/ns
```
Comment (trantanen) [inline samples/cellular/modem_shell/sample.yaml:11]: Modem shell should be handled in PR #15225.

File: samples/cellular/location/sample.yaml [hunk 3]
```diff
@@ -72,12 +72,16 @@ tests:
   sample.cellular.location.thingy91x:
     sysbuild: true
     build_only: true
+    integration_platforms:
+      - thingy91x/nrf9151/ns
```
Comment (trantanen) [inline samples/cellular/location/sample.yaml:76]: Location sample is handled in PR #15266. If you want, you can copy those minor changes into this PR or drop location sample from this PR.

Comment (maxd-nordic) [reply samples/cellular/location/sample.yaml:76]: I'll drop my changes here.

**Comment by trantanen** [comment]
You should get document build succeed after `sample_board_rows.txt` changes in PR #15225

File: applications/asset_tracker_v2/sample.yaml [hunk 4]
```diff
@@ -15,6 +15,7 @@ tests:
     integration_platforms:
       - nrf9160dk/nrf9160/ns
       - thingy91/nrf9160/ns
+      - thingy91x/nrf9151/ns
```
[the remark this answers is not in the document]
Comment (maxd-nordic) [reply applications/asset_tracker_v2/sample.yaml:18]: not yet

**Review by trantanen — APPROVED** [review]

**Review by divipillai — APPROVED** [review]
doc update (minor) + release notes would be covered in a separate PR for these samples.

**Comment by shanthanordic** [comment]
CI is very RED, moving it out of NCS 2.7.0 @maxd-nordic

**Comment by maxd-nordic** [comment]
> CI is very RED, moving it out of NCS 2.7.0 @maxd-nordic

This is a documentation PR. If CI is RED in general, it should not block this PR. @shanthanordic

**Comment by divipillai** [comment]
@shanthanordic : This PR was requested to update the "integration_platforms" field, so that the requirements table of the sample docs are update with the Thingy:91 support. https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/applications/serial_lte_modem/doc/slm_description.html#requirements

Is it okay to add a CI-disable label to this PR, considering the above? or should all the tests pass for this PR?

**Comment by jorgenmk** [comment]
This PR adds some new build configurations to twister stage, so all tests should pass. The failing test here is thingy91 + carrier build, which had problems earlier this week. Perhaps a rebase will fix?

**Comment by jtguggedal** [comment]
Pushed with cherry-picked [commit](https://github.com/nrfconnect/sdk-nrf/pull/15569/commits/a55367a5f683ec18127845a5245e1679caf2621c) from #15569 that disables the problematic testcase in CI, as it's already broken.
FYI @nordic-hani

**Comment by jtguggedal** [comment]
> CI is very RED, moving it out of NCS 2.7.0 @maxd-nordic

@shanthanordic CI is green now, so this can be merged for the release

**Pull request merged**